### DISCUSSION TITLE
Is there any better ways to upload a huge pipeline to mlflow?

### DISCUSSION THREAD

**Discussion by kimminw00** [mlflow/mlflow#12620] [Q&A] [answered]
```python
import mlflow
from transformers import pipeline

pipe = pipeline("text-generation", model="upstage/SOLAR-0-70b-16bit")

with mlflow.start_run():
    mlflow.transformers.log_model(
        transformers_model=pipe,
        artifact_path="my_pipeline",
    )
```
When I run this script, it loads all components into memory at the same time.
So python process is being killed by the OOM killer.
I don't think there is no reason to load the whole components into memory.
Is there any better ways to log a huge pipeline as a whole into mlflow?
(without individually logging components into mlflow)

**Comment by B-Step62** [comment, marked as answer]
@kimminw00 Unfortunately, we only support in-memory logging at this moment. However, we are actively working on adding a pathway to log a pipeline without loading it into memory. There are a few functionalities need to be dropped e.g. requirement inference, but if things go well the capability can be released in near future.
